README.md
# News Mangement with c++ using oop and Data structure .

##Features :
  ### -Login to an account(if it exists) or Register a new one.
  ### -Admin
      Post news to categories
      Delete news from specific category
      Update existing news
  ### -User
      Display latest news descendingly according     to rates
      Show news depending on specific category
      Each news has only one rate for each user 
      Spam News
      Comment on News  


News.txt
11
supreme court makes it more difficult for prisoners to argue they had ineffective counsel
Ariane de Vogue and Devan Cole
Washington (CNN)The Supreme Court said Monday that state prisoners may not present new evidence in federal court in support of a claim that their post-conviction counsel in state court was ineffective in violation of the Constitution.
2022/05/23
0.000000
0
0
16
it's going to be hot with a chance of blackouts
Zachary B. Wolf
It's going to be hot out there
2022/05/21
0.000000
0
0
21
joe biden just keeps stepping in it
Chris Cillizza
Within hours, his White House aides insisted that the President's comments did not constitute any official US policy change, despite the fact that Biden has suggested at other points in recent months that the US would defend Taiwan if the country was attacked by China.
2022/05/23
3.300000
1
0
31
why you shouldn't believe this latest donald trump promise
Chris Cillizza
Remember that Trump directly linked Twitter to his 2016 victory. "I doubt I would be here if it weren't for social media, to be honest with you," he said in 2017. "When somebody says something about me, I am able to go bing, bing, bing and I take care of it. The other way, I would never be get the word out."
2020/05/01
0.000000
0
0
13
10 most famous paintings in the world
Forrest Brown
It should come as no surprise that the most famous painting in the world is that mysterious woman with the enigmatic smile. But that's one of the few certainties about this work of art.
2019/09/22
0.000000
0
0
23
the starry night
Forrest Brown
The comparatively abstract painting is the signature example of van Gogh's innovative and bold use of thick brushstrokes. The painting's striking blues and yellows and the dreamy, swirling atmosphere have intrigued art lovers for decades.
2019/02/11
0.000000
0
0
33
the scream
Forrest Brown
Much like the case of "Mona Lisa," daring thefts (1994 and 2004) of the two painting versions of "The Scream" helped elevate the public's awareness of the artworks. (Both were eventually found).
2018/02/19
0.000000
0
0
14
hannah rankin opponent alejandra ayala out of induced coma
BBC
The Scot retained her WBA and IBO super-welterweight titles with a 10th-round knockout on 13 May.
2022/05/23
0.000000
0
0
24
emma raducanu wins thriller in roland garros first round
Jonathan Jurejko
Emma Raducanu showed fight and spirit to come back from a set down and win her senior debut at the French Open against Czech teenager Linda Noskova.
2020/01/17
0.000000
0
0
34
chelsea deal 'still has major hurdles to overcome' whitehall source tells bbc sport
Dan Roan
It is understood that, because owner Roman Abramovich has a Portuguese passport, the takeover needs to be licensed by authorities in that country too.
2022/05/24
0.000000
0
0
12
economic growth in the uae drives surge in property deals
Economy Middle East
Economic growth, positive market sentiment, and the influx of new capital resulted in a surge in real estate transaction values and volumes; as such, sales prices recorded a significant spike, particularly in regard to villas, according to a new report conducted by property management firm Atesco. 
2022/05/23
5.000000
1
0
22
wef president lauds saudi reforms
Economy Middle East
Borge Brende also disclosed that the Geneva-based organization is considering taking the Middle East and North Africa (MENA) regional summit to Riyadh.
2018/02/10
3.300000
1
0


News.h
#pragma once
#include <iostream>
#include "Admin.h"
#include <fstream>
#include <string>
#include <vector>
#include <assert.h>
#include "User.h"
#include <unordered_map>
using namespace std;

struct  Details {

	string Title;
	string Writer;
	string Discrepition;
	string Date;
	string string_rate;
	string count;
	string spam;
};

class News
{

	string comment;
	string userComment;
	int count=0;
	float Rate;
	int total=0;
	int type = 0 ; 
	int NumberOfRate =0;
	int NumberOfSpam = 0;
public :
	void ChangeSpam(string);
	Details detail;
	string	string_type;
	News(void);
	vector<Details> ReadNews();
	unordered_map<int, vector<Details>> ReadCatogNews();
	bool ReadCertainNews(string);
	void WriteNews(int , string[]);
	void WriteComment(string);
	float CalculateRate(string , float , string );
	unordered_map<string , vector<string> > ReadComment();
	//vector<string>ReadComment(string title);
	int EndNumberOfNewsInCatogry(int);
	void ChangeRate(string , float);
};


User.h
#pragma once
#include "News.h"
#include "Admin.h"
#include<iostream>
#include<fstream>
#include<string>
#include<vector>
#include <unordered_map>
using namespace std;
struct spamDetails {
	string use;
	string title;
};
struct  Details;
class User
{
	vector<string> spamNews;
	int choice;
	float rate;
	int count = 0;
	unordered_map<string, vector<string> >  comments;
	unordered_map<int, vector<Details>> specificNews;
	unordered_map<string, vector<string>> msam;
	
public:
	string tit;
	string newsnumber;
	string user;
	string cata;
	void login();
	void registration();
	void intialization();
	void forget_pass();
	bool checkRateAll(string);
	void OptionRateAll();
	void RateAll();
	void Display_Latest_News();
	void Show();
	void Spam();
	void OptionSpam();
	void DisplayComment();
	void op();
	unordered_map<string, vector<string>> pam();
};


News.cpp
#include "News.h"
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <assert.h>
#include <sstream>
#include <algorithm>
#include <unordered_map>
using namespace std;
User user;

News::News() {
	Rate = 0;
	count=0;

}

vector<Details>News::ReadNews() {
	ifstream input;
	input.open("News.txt", ios::app);
	assert(!input.fail());
	vector<Details> details;
	while (getline(input ,string_type) && getline(input, detail.Title) && getline(input, detail.Writer) &&
		getline(input, detail.Discrepition)&& getline(input, detail.Date )&& 
		getline(input, detail.string_rate) &&
		getline(input, detail.count) &&
		getline(input, detail.spam))
	{

		details.push_back(detail);

	} input.close();
	return details;
}

unordered_map<int, vector<Details>> News::ReadCatogNews () {
	ifstream input;
	input.open("News.txt", ios::app);
	assert(!input.fail());
	vector<Details> details;
	unordered_map<int, vector<Details>> news;
	while (getline(input, string_type) &&
		getline(input, detail.Title) && 
		getline(input, detail.Writer) &&
		getline(input, detail.Discrepition) && 
		getline(input, detail.Date) &&
		getline(input, detail.string_rate) &&
		getline(input, detail.count) &&
		getline(input, detail.spam))
	{
		
		stringstream geek(string_type);
		geek >> type;
		type = type % 10;
		if (news.find(type) != news.end()) {
			details = news.at(type);
			news.erase(type);
		}
		details.push_back(detail);
		news.insert(make_pair(type, details));
		details.clear();
	}
	
	input.close();
	return news;
}

bool News::ReadCertainNews(string titl) {

	ifstream input;
	input.open("News.txt");

	while (getline(input, string_type) && 
		getline(input, detail.Title) && 
		getline(input, detail.Writer) &&
		getline(input, detail.Discrepition) && 
		getline(input, detail.Date) && 
		getline(input, detail.string_rate) &&
		getline(input, detail.count) &&
		getline(input, detail.spam))
	{
		transform(titl.begin(), titl.end(), titl.begin(), ::tolower);
		if (titl == detail.Title ) {
			
			input.close();
			return true;
		}

	} 
	input.close();
	return false;
//	cout << detail.Title << "        " << detail.Writer << "     " << detail.Discrepition << "        " << detail.Date << endl;
}

float News::CalculateRate(string oldRate , float rate , string counter) {
	float old = stof(oldRate);
	stringstream geek(counter);
	geek >> NumberOfRate;
	++NumberOfRate;
	detail.count = to_string(NumberOfRate);
	return ( (NumberOfRate-1.0)/ NumberOfRate )* old +(1.0 / NumberOfRate )* rate;
}

void News::ChangeRate( string titl , float rate) {

	bool isFoundTitle = ReadCertainNews(titl);
	cout << isFoundTitle << endl;
	if (isFoundTitle == true ) {
		ofstream tempfile;
		tempfile.open("temp.txt");
		ifstream input;
		input.open("News.txt");
		while (getline(input, string_type) &&
			getline(input, detail.Title) &&
			getline(input, detail.Writer) &&
			getline(input, detail.Discrepition) && 
			getline(input, detail.Date )&&
			getline(input, detail.string_rate) &&
			getline(input, detail.count) &&
			getline(input, detail.spam))
		{
			transform(titl.begin(), titl.end(), titl.begin(), ::tolower);
			if (titl != detail.Title)
			{
				tempfile << string_type << "\n"
					<< detail.Title << "\n"
					<< detail.Writer << "\n"
					<< detail.Discrepition << endl
					<< detail.Date << endl
					<< detail.string_rate << endl
					<< detail.count << endl
					<< detail.spam << endl;
			
			}
			else
			{
				float newRate = CalculateRate(detail.string_rate, rate, detail.count);
				if (newRate <= 2) {
					continue;
				}
				cout << newRate << endl;
				tempfile << string_type << "\n"
					<< detail.Title << "\n"
					<< detail.Writer << "\n"
					<< detail.Discrepition << endl
					<< detail.Date << endl
					<< to_string(newRate)<< endl
					<< detail.count << endl
					<< detail.spam << endl;
				
			}

		}
		input.close();
		tempfile.close();
		remove("News.txt");
		rename("temp.txt", "News.txt");
		cout << "Done \n";
	}
	else {
		cout << "Sorry, The title that you search about isn't found ." << endl;
		user.RateAll();
		return;
	}
}

void News::WriteNews(int Type , string detail[]) {
	ofstream input("News.txt", ios::app);
	assert(!input.fail());
	transform(detail[0].begin(), detail[0].end(), detail[0].begin(), ::tolower);
	input << to_string(Type) << endl
		<< detail[0] << endl
		<< detail[1] << endl
		<< detail[2] << endl
		<< detail[3] << endl
		<< to_string(Rate) << endl
		<< to_string(count) << endl 
		<< to_string(NumberOfSpam) << endl;
	system("cls");
	input.close();
}

int News::EndNumberOfNewsInCatogry(int cato) {
	ifstream input;
	input.open("News.txt", ios::app);
	assert(!input.fail());
	while (getline(input, string_type)&&
		getline(input, detail.Title)&&
		getline(input, detail.Writer)&&
		getline(input, detail.Discrepition)&&
		getline(input, detail.Date)&&
		getline(input, detail.string_rate)&& 
		getline(input, detail.count) &&
		getline(input, detail.spam)) {

		stringstream geek(string_type);
		geek >> type;
		if (cato == (type % 10) ){
			
			count = type; 
		}
	}
	return count ;
}

void News::WriteComment(string user) {
	string title, comments;
	cout << "please enter title " << endl;
	getline(cin >> ws, title);
	bool isFound = ReadCertainNews(title);
	bool isComment = false ;
	ifstream input("Comments.txt", ios::app);
	assert(!input.fail());
	ofstream tempfile;
	tempfile.open("tempfile.txt");

	if (isFound) {
		cout << "please enter comment " << endl;
		getline(cin, comments);
		while (getline(input, string_type)  ) {
			tempfile << string_type << "\n";
			while ( getline(input, userComment))
			{
				if (userComment != "1")
				{
					getline(input, comment);
					tempfile << userComment << "\n" << comment << "\n";
				}
				else
				{
					
					if (string_type == title) {
						tempfile << user << " comment :\n"  << comments << "\n";
						isComment = true;
					}
					tempfile << userComment << "\n";
					break;
				}
			}
		}
		if(!isComment)
			tempfile  << title << "\n" << user << " comment :\n" << comments << "\n" << "1" << "\n";
		input.close();
		tempfile.close();
		remove("Comments.txt");
		rename("tempfile.txt", "Comments.txt");
		
	}
	else 
	{
		cout << "Sorry, The title that you search about isn't found ." << endl;
		cout << " press 1 To continue to comment on another news or press 2 if not " << endl;
		string choose;
		cin >> choose;
		if (choose == "1") {
			WriteComment(user);
			return;
		}
		else if (choose == "2") {
			return;
		}
		else {
			cout << "Please , Enter 1 or 2 only" << endl;
			cin >> choose;
			WriteComment(user);
		}
	}
	system("cls");
}

unordered_map<string, vector<string>> News::ReadComment() {
	ifstream input;
	input.open("Comments.txt", ios::app);
	assert(!input.fail());
	vector<string> details;
	unordered_map<string, vector<string>> comments ;
	while (getline(input, string_type) ) {
		while ( getline(input, userComment) )
		{
			if (userComment == "1")
			{
				break;
			}
			else
			{
				getline(input, comment);
				details.push_back(userComment);
				details.push_back(comment);
			
			}
		}
		if(!details.empty())
			comments.insert(make_pair(string_type, details));
		details.clear();
	}
	input.close();
	return comments;
}

//vector<string> News::ReadComment(string title) {
//	ifstream input;
//	input.open("Comments.txt", ios::app);
//	assert(!input.fail());
//	vector<string> details;
//	transform(title.begin(), title.end(), title.begin(), ::tolower);
//	while (getline(input, string_type)) {
//		while (getline(input, userComment))
//		{
//			if (userComment == "1")
//			{
//				break;
//			}
//			else
//			{
//				getline(input, comment);
//				if (string_type == title) {
//					details.push_back(userComment);
//					details.push_back(comment);
//
//				}
//			}
//		}
//	}
//	input.close();
//	return details;
//}

void News::ChangeSpam(string titl) {

	bool isFoundTile = ReadCertainNews(titl);
	cout << isFoundTile << endl;
	if (isFoundTile == true) {
		ofstream tempfile;
		tempfile.open("temp.txt");
		ifstream input;
		input.open("News.txt");
		while (getline(input, string_type) &&
			getline(input, detail.Title) &&
			getline(input, detail.Writer) &&
			getline(input, detail.Discrepition) &&
			getline(input, detail.Date) &&
			getline(input, detail.string_rate) &&
			getline(input, detail.count) &&
			getline(input, detail.spam))
		{
			transform(titl.begin(), titl.end(), titl.begin(), ::tolower);
			if (titl != detail.Title)
			{
				tempfile << string_type << "\n"
					<< detail.Title << "\n"
					<< detail.Writer << "\n"
					<< detail.Discrepition << endl
					<< detail.Date << endl
					<< detail.string_rate << endl
					<< detail.count << endl
					<< detail.spam << endl;
			}
			else
			{
				stringstream geek(detail.spam);
				geek >> NumberOfSpam;
				NumberOfSpam++;
				tempfile << string_type << "\n"
					<< detail.Title << "\n"
					<< detail.Writer << "\n"
					<< detail.Discrepition << endl
					<< detail.Date << endl
					<< detail.string_rate << endl
					<< detail.count << endl
					<< to_string(NumberOfSpam) << endl;
				
			}

		}
		input.close();
		tempfile.close();
		remove("News.txt");
		rename("temp.txt", "News.txt");
		cout << "Done \n";
	}
	else {
		cout << "Sorry, The title that you search about isn't found ." << endl;
		cout << " press 1 To continue to Spam news or press 2 if not " << endl;
		int choose;
		cin >> choose;
		if (choose == 1) {
			user.Spam();
		}
		else if (choose == 2) {
			return;
		}
		else
		{
			cout << "Please , Enter 1 or 2 only" << endl;
		}
	}
}


Admin.cpp
#include "Admin.h"
//#include "News.h"
#include<iostream>
#include<fstream>
#include<string>
#include <algorithm>
#include<vector>
using namespace std;
User re;
Admin::Admin()
{
	category = "";
	title = "";
	name = "";
	story = "";
	countpol = 0;
	counteco = 0;
	countart = 0;
	countspo = 0;
	countinter = 0;
	countsoc = 0;
}

int Admin::NumberOfNews(string cato , int PostOrUpdate) {
	News news;
	int value = 0;
	transform(cato.begin(), cato.end(), cato.begin(), ::tolower);
	if ( cato == "political" )
	{
		value = news.EndNumberOfNewsInCatogry(1);
		if (value == 0)
			return 11;
		else {
			countpol = 1 + ((value / 10) + 1) * 10;
			return countpol;
		}
	}
	else if (cato == "economic" ) {
		value = news.EndNumberOfNewsInCatogry(2);
		if (value == 0)
			return 12;
		else {
			counteco = 2+ ((value / 10) + 1) * 10;
			return counteco;
		}
	}
	else if (cato == ("art")) {
		value = news.EndNumberOfNewsInCatogry(3);
		if (value == 0)
			return 13;
		else {
			countart = 3 + ((value / 10) + 1) * 10;
			return countart;
		}
	}
	else if (cato == "sports") {
		value = news.EndNumberOfNewsInCatogry(4);
		if (value == 0)
			return 14;
		else {
			countspo = 4 + ((value / 10) + 1) * 10;
			return countspo;
		}
	}
	else if (cato == ("social")) {
		
		value = news.EndNumberOfNewsInCatogry(5);
		
		if (value == 0)
			return 15;
		else {
			countsoc = 5 +( (value / 10)+1 ) * 10;
			return countsoc;
		}
	}
	else if (cato == "weather") {
		value = news.EndNumberOfNewsInCatogry(6);
		
		if (value == 0)
			return 16;
		else {
			cout << countinter << endl;
			countinter = 6 + ((value / 10) + 1) * 10;
		
			return countinter;
		}
	}
	else {
		cout << "Sorry, The category that you search about isn't found ."<< endl ;
		if (PostOrUpdate == 0)
			post();
		else if (PostOrUpdate == 1)
			update();
		return 0;
	}
	
}

//
//void Admin::post() {
//	News news;
//	cout << "enter the category of the news: \n";
//	getline(cin>> ws, category);
//	int value = NumberOfNews(category, 0);
//	if (value == 0)
//		return;
//	cout << "enter the title of the news: \n";
//
//	getline(cin, title);
//	detail[0] = title;
//	//cout << title<< endl;
//	cout << "enter the name of the writer:\n";
//
//	getline(cin, name);
//	detail[1] = name;
//	//cout << name<<endl;
//	cout << "enter the story of the news: \n";
//
//	getline(cin, story ) ;
//	detail[2] = story;
//	//cout << story<< endl;
//	cout << "enter the Date of the news: \n";
//	
//	getline(cin, Date );
//	detail[3] = Date;
//	//cout << Date << endl;
//	news.WriteNews( value , detail);
//
//	cout << " data saved:\n";
//	
//}

void Admin::post() {
	News news;
	Newsdetails newsdetails;
	cout << "enter the category of the news: \n";
	getline(cin >> ws, newsdetails.category);
	int value = NumberOfNews(newsdetails.category, 0);
	if (value == 0)
		return;
	cout << "enter the title of the news: \n";

	getline(cin, newsdetails.title);
	detail[0] = newsdetails.title;
	//cout << title<< endl;
	cout << "enter the name of the writer:\n";

	getline(cin, newsdetails.name);
	detail[1] = newsdetails.name;
	//cout << name<<endl;
	cout << "enter the story of the news: \n";

	getline(cin, newsdetails.story);
	detail[2] = newsdetails.story;
	//cout << story<< endl;
	cout << "enter the Date of the news: \n";

	getline(cin, newsdetails.Date);
	detail[3] = newsdetails.Date;
	//cout << Date << endl;
	news.WriteNews(value, detail);
	cout << " data saved:\n";

}

void Admin::delete_category() {
	News news;
	string titl;
	cout << "enter the title: \n";
	cin >> titl;
	bool r= news.ReadCertainNews(titl);
	if (r == true) {
		cout << "if you want to delete a record press( y):\n";
		char option;
		cin >> option;
		if (option == 'y' || option == 'Y') {
			ofstream tempfile;
			tempfile.open("temp.txt");
			ifstream read;
			read.open("News.txt");
			while (getline(read, category) &&
				getline(read, title) &&
				getline(read, name) &&
				getline(read, story) &&
				getline(read, Date) &&
				getline(read, rate) &&
				getline(read, news.detail.count) &&
				getline(read, news.detail.spam))
			{
				transform(titl.begin(), titl.end(), titl.begin(), ::tolower);
				if (titl != title)
				{
					tempfile << category << "\n"
						<< title << "\n"
						<< name << "\n"
						<< story << endl
						<< Date << endl
						<< rate << endl
						<< news.detail.count << endl
						<< news.detail.spam << endl;

				}

			}
			read.close();
			tempfile.close();
			remove("News.txt");
			rename("temp.txt", "News.txt");
			cout << "data is removed:\n";
		}
	}
	else 
	{
		cout << "Sorry, The title that you search about isn't found ." << endl;
		cout << " press 1 To continue to delete news or press 2 if not " << endl;
		int choose; 
		cin >> choose;
		if (choose == 1) {
			delete_category();
		}
		else if (choose == 2) {
			return;
		}
		else {
			cout << "Please , Enter 1 or 2 only" << endl;
		}
	}
}

void Admin::update() {
	News news;
	string titl;
	cout << "enter the title: \n";
	cin >> titl;
	bool r = news.ReadCertainNews(titl);
	if (r) {
		char option;
		cout << "if you want to Update a record press( y):\n";
		cin >> option;
		if (option == 'y' || option == 'Y') {
			ofstream tempfile;
			tempfile.open("temp.txt");
			ifstream read;
			read.open("News.txt");
			cout << "enter the category of the news: " << endl;
			getline(cin >> ws, category);
			int value = NumberOfNews(category, 1);
			if (value > 0) {
				cout << "enter the title of the news: " << endl;
				getline(cin, title);
				//cout << title << endl;
				cout << "enter the name of the writer:\n";
				getline(cin, name);
				//cout << name << endl;
				cout << "enter the story of the news: \n";
				getline(cin, story);
				//cout << story << endl;
				cout << "enter the Date of the news: \n";
				getline(cin, Date);
				//cout << Date << endl;
				while (getline(read, news.string_type) && 
					getline(read, news.detail.Title) && 
					getline(read, news.detail.Writer) &&
					getline(read, news.detail.Discrepition) && 
					getline(read, news.detail.Date) &&
					getline(read, news.detail.string_rate)&& 
					getline(read, news.detail.count) &&
					getline(read, news.detail.spam))
				{
					transform(titl.begin(), titl.end(), titl.begin(), ::tolower);
					if (titl != news.detail.Title)
					{
						tempfile << news.string_type << "\n"
							<< news.detail.Title << "\n"
							<< news.detail.Writer << "\n"
							<< news.detail.Discrepition << endl
							<< news.detail.Date << endl
							<< news.detail.string_rate << endl
							<< news.detail.count << endl
							<< news.detail.spam << endl;
						
					}
					else
					{
						transform(title.begin(), title.end(), title.begin(), ::tolower);
						tempfile << to_string(value) << "\n"
							<< title << "\n"
							<< name << "\n"
							<< story << endl
							<< Date << endl
							<< news.detail.string_rate << endl
							<< news.detail.count << endl
							<< news.detail.spam << endl;
						
					}
				}
			}
			read.close();
			tempfile.close();
			remove("News.txt");
			rename("temp.txt", "News.txt");
			cout << "data is Updated .\n";
		}
	}
	else
	{
		cout << "Sorry, The title that you search about isn't found ." << endl;
		cout << " press 1 To continue to update news or press 2 if not " << endl;
		int choose;
		cin >> choose;
		if (choose == 1) {
			update();
		}
		else if (choose == 2) {
			return;
		}
		else {
			cout << "Please , Enter 1 or 2 only" << endl;
		}
	}
}

void Admin::readdata() {
	News news;
	vector<Details> st = news.ReadNews();
	
	cout << "****** THE NEWS ******* \n";
	for (int i = 0 ; i < st.size() ; i++) {
		cout << st[i].Title << "\n";
		cout << st[i].Writer << "\n";
		cout << st[i].Discrepition << "\n";
		cout << st[i].Date << "\n";
		cout << st[i].string_rate << "\n";
		
	}
	
}
void Admin::options() {


	cout << "1:To post a new \n";
	cout << "2:To update a new \n";
	cout << "3:To delete data a new \n";
	cout << "4:To read data a new \n";
	cout << "5:To logout \n";
	cout << "enter your option :\n";
	cin >> option;
	switch (option)
	{
	case 1:
		post();
		break;
	case 2:
		update();
		break;
	case 3:
		delete_category();
		break;
	case 4:
		readdata();
		break;
	case 5:
		system("cls");
		re.intialization();
		break;
	default:
		cout << "invalid option: \n";
		break;
	}
	Choice();
}

void Admin::Choice()
{
	cout << "If you want to choose another choice press --> 7 or 8 to log out" << endl;
	cin >> option;
	if (option == 7) {
		options();
		system("cls");
	}
	else if (option == 8)
	{
		re.intialization();
		system("cls");
	}

}
